targets/XPlatCppSdk/source/code/source/playfab/PlayFabIXHR2HttpPlugin.cpp

```diff
@@ -0,0 +1,254 @@
+
+        // Header 2
+        HttpHeaderInfo acceptHeader;
+        acceptHeader.wstrHeaderName = L"Accept";
+        acceptHeader.wstrHeaderValue = L"application/json";
+        headers.push_back(std::move(acceptHeader));
+
+        // Header 3
+        HttpHeaderInfo playFabSdkHeader;
+        playFabSdkHeader.wstrHeaderName = L"X-PlayFabSDK";
+        playFabSdkHeader.wstrHeaderValue = std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end());
+        headers.push_back(std::move(playFabSdkHeader));
+
+        // Header 4
+        HttpHeaderInfo reportErrorAsSuccessHeader;
+        reportErrorAsSuccessHeader.wstrHeaderName = L"X-ReportErrorAsSuccess";
+        reportErrorAsSuccessHeader.wstrHeaderValue = L"true";
+        headers.push_back(std::move(reportErrorAsSuccessHeader));
+
+        auto reqHeaders = reqContainer.GetHeaders();
+
+        if (reqHeaders.size() > 0)
+        {
+            for (auto const &obj : reqHeaders)
+            {
+                if (obj.first.length() != 0 && obj.second.length() != 0) // no empty keys or values in headers
+                {
+                    HttpHeaderInfo hInfo;
+                    hInfo.wstrHeaderName = std::wstring(obj.first.begin(), obj.first.end());
+                    hInfo.wstrHeaderValue = std::wstring(obj.second.begin(), obj.second.end());
+                    headers.push_back(std::move(hInfo));
+                }
+            }
+        }
+    }
+
+    void PlayFabIXHR2HttpPlugin::ExecuteRequest(std::unique_ptr<CallRequestContainer> requestContainer)
+    {
+        CallRequestContainer& reqContainer = *requestContainer;
+
+        // Create request
+        HttpRequest postEventRequest;
+        
+        // Setup headers
+        std::vector<HttpHeaderInfo> headers;
+        SetupRequestHeaders(reqContainer, headers);
+
+        // Setup url
+        std::string urlString = PlayFabSettings::GetUrl(reqContainer.GetUrl(), PlayFabSettings::requestGetParams);
+        std::wstring url(urlString.begin(), urlString.end());
+        
+        // Setup payload
+        std::string payload = reqContainer.GetRequestBody();
+
+        // Send post request
+        postEventRequest.Open(
+            L"POST",
+            url,
+            headers,
+            payload);
+
+        // Wait for the request to complete
+        postEventRequest.WaitForFinish();
+
+        const HRESULT res = postEventRequest.GetResult();
+        const DWORD status = postEventRequest.GetStatus();
+        auto response = postEventRequest.GetData();
+        reqContainer.responseString = std::string(response.begin(), response.end());
+
+        // 401 is a special case where the status does not bubble up and we have to parse it from the HRESULT
+        if (status == 401)
+        {
+            reqContainer.errorWrapper.HttpCode = 401;
+            reqContainer.errorWrapper.HttpStatus = "Access denied";
+
+            // TODO sangarg : Fix the error code here
```

Comment:
@Vicodex : Please check the error code here. Should we add a new error code for 401? We are currently using PlayFabErrorConnectionTimeout.